workflow_task/models/task.py

```diff
@@ -73,6 +73,17 @@ def _select_objects(self):
                                  compute='_get_action_ids')
     pretty_res_type = fields.Char(compute='_get_pretty_res_type')
 
+    
+    def fields_get(self, cr, user, allfields=None, context=None,
+                   write_access=True, attributes=None):
+        res = super(Task, self).fields_get(
+            cr, user, allfields, context, write_access, attributes)
+        # remove ref_object from searchable field into the advanced search
+        # since the field to use is ref_object_name
+        if 'ref_object' in res:
+            res['ref_object']['searchable'] = False
```

Comment:
@adrienpeiffer unfortunately no :sob:  `searchable`is defined as a python property on `fields.Field` https://github.com/odoo/odoo/blob/9.0/openerp/fields.py#L675
You can see at line 125 how odoo compute the list of attributes returned by the function field_get https://github.com/odoo/odoo/blob/9.0/openerp/fields.py#L125
The only solution I've found is override `fields_get` to change the value returned to the JS by the server